### PR TITLE
Add notebook explaining classical shadow estimation

### PR DESCRIPTION
**Pull request by julien-bremont**
This PR adds a Jupyter notebook which purpose is to expose the concept of classical shadow estimation using Pulser. It gives a general idea of the method, implements it and uses it for the estimation of the ground-state energy of the H2 molecule using a VQS algorithm, programmed in Pulser.

### CODE REVIEW

File: tutorials/applications/Shadow estimation for VQS.ipynb [hunk 1]
```diff
@@ -0,0 +1,1460 @@
+{
```
Comment (lhenriet) [inline tutorials/applications/Shadow estimation for VQS.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #21.        N = 20
</pre><p>What is N?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (lhenriet) [inline tutorials/applications/Shadow estimation for VQS.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #11.    dist = np.zeros(5)
</pre><p>For conciseness, maybe create a shadow_sizes list =[100, 1000, 2000, 5000, 10000]. Re-use it in the plt.plot line below.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (lhenriet) [inline tutorials/applications/Shadow estimation for VQS.ipynb:1]: <p>"Feeding these observables and chosen Pauli measurements {P_i} as input, the algorithm aims at optimizing a certain cost function. This function [...]"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (lhenriet) [inline tutorials/applications/Shadow estimation for VQS.ipynb:1]: <p>I have a display pbm here.</p><p> </p><p>Also, I do not find the sentence "[...], meaning that on each qubit that o acts non-trivially upon, it must act with the same Pauli matrix as p" clear enough. Reformulate ?</p><p> </p><p>"[...] for each qubit that o acts non-trivially upon, the action is the same Pauli in o and p" or something like that (not convinced by my own proposition).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (lhenriet) [inline tutorials/applications/Shadow estimation for VQS.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #9.        # Hyperparameters : see 
</pre><p>?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (lhenriet) [inline tutorials/applications/Shadow estimation for VQS.ipynb:1]: <p>[Math Processing Error] ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (lhenriet) [inline tutorials/applications/Shadow estimation for VQS.ipynb:1]: <p>classical</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='66'/>

Comment (lhenriet) [inline tutorials/applications/Shadow estimation for VQS.ipynb:1]: <p>Great notebook. Would it be possible to mention the total number of measurements necessary for this VQS procedure on H_2 ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pasqal-io/Pulser/pull/252/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='84'/>

Comment (julien-bremont) [reply tutorials/applications/Shadow estimation for VQS.ipynb:1]: <p>I wrote down the theoretical number of shadow size per cluster on the line above, but this makes simulations run too slowly on a laptop. After some trial and error I figured <code>N=20</code>  was good enough.</p>

Comment (lhenriet) [reply tutorials/applications/Shadow estimation for VQS.ipynb:1]: <p>Great</p>

**Review by lhenriet — APPROVED** [review]
Looks good.

**Pull request merged**